src/static/utils/darkable-image.js

```diff
@@ -0,0 +1,41 @@
+const darkmodeMedia = '(prefers-color-scheme: dark)';
+
+class DarkmodeIcon extends HTMLElement {
+    connectedCallback() {
+        const isDarkMode = window.matchMedia(darkmodeMedia).matches;
+        this.img = document.createElement('img');
+        this.img.src = renameFile(this.src, isDarkMode);
+        this.img.alt = this.alt;
+        for (let attr of this.attributes) {
+            if (attr.name === 'class') continue;
+            this.img.setAttribute(attr.name, attr.value);
+        }
+
+        this.appendChild(this.img);
+        this.classList.add('ui-img-dark');
+    }
+
+    updateImageSource(isDarkMode) {
+        this.img.src = renameFile(this.img.src, isDarkMode);
+    }
+}
+
+customElements.define('atb-img-dark', DarkmodeIcon);
+
+const mql = window.matchMedia(darkmodeMedia);
+updateAllDarkmodeIcons(mql);
+mql.addEventListener('change', updateAllDarkmodeIcons);
+
+function updateAllDarkmodeIcons(e) {
+    const elements = Array.from(document.querySelectorAll('atb-img-dark'));
+    elements.forEach(function (img) {
+        img.updateImageSource(e.matches);
+    });
+}
+
+function renameFile(filename, isDarkMode) {
+    if (!isDarkMode) {
+        return filename.replace('-dark.', '.');
+    }
+    return filename.replace(/\.(\w{3})$/, '-dark.$1');
```

Comment:
Tillate 4 chars også? Ser for meg at det en eller annen gang kommer til å bli feilsøk på hvorfor det ikke funger på et bilde i jpeg-format. Eller er dette i hovedsak ment for svg?

Reply:
Kunne egentlig ha tillatt flere formater sånn sett. Bare jeg som prøvde å være litt restriktiv. Men jpeg er et godt poeng.

Reply:
Kunne whitelisting av aktuelle bildeformater fungert her i en eller annen form, @mikaelbr?

Reply:
Ja, jo. Men konsekvensene blir på en måte det samme tenker jeg. Om det ikke treffer vil default (ikke-`-dark`) vises og det vil være noe tvetydighet om hvorfor det ikke fungerer.